### ISSUE TITLE
test-runner rewriting `null` oddly

### ISSUE THREAD

**Issue by mlucy** [rethinkdb/rethinkdb#3580]
I tried to add the following test:
```
    - cd: tbl.get(0).changes().limit(1)
      ot: [null]
```

And I got the following errors:
```
~/rethinkdb/test/rql_test $ ./test-runner polyglot/changefeeds/basic.*
Using rethinkdb binary /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb
        javascript interpreter: /usr/local/bin/node, driver: /home/ssd2/mlucy/rethinkdb/drivers/javascript
        python 2.6 interpreter: /usr/bin/python2.6, driver: /home/ssd2/mlucy/rethinkdb/drivers/python
        ruby 1.9 interpreter: /usr/bin/ruby1.9, driver: /home/ssd2/mlucy/rethinkdb/drivers/ruby
== Starting: polyglot/changefeeds/basic.js (T+ 2.2 sec)
== Starting: polyglot/changefeeds/basic.py2.6 (T+ 2.7 sec)
>>> Failed polyglot/changefeeds/basic.js with exit code 1 after 1.3 sec (T+ 3.8 sec)

Using RethinkDB client from: /home/ssd2/mlucy/rethinkdb/build/packages/js
polyglot/changefeeds/basic.js/1
[None]
stack: ReferenceError: None is not defined
    at eval (eval at runTest (/home/ssd2/mlucy/rethinkdb/test/rql_test/build/polyglot.changefeeds.basic.js:243:44), <anonymous>:1:2)
    at runTest (/home/ssd2/mlucy/rethinkdb/test/rql_test/build/polyglot.changefeeds.basic.js:243:39)
    at afterArray (/home/ssd2/mlucy/rethinkdb/test/rql_test/build/polyglot.changefeeds.basic.js:342:21)
    at run_callback (/home/ssd2/mlucy/rethinkdb/test/rql_test/build/polyglot.changefeeds.basic.js:404:25)
    at /home/ssd2/mlucy/rethinkdb/test/rql_test/build/polyglot.changefeeds.basic.js:294:112
    at tryCatch2 (/home/ssd2/mlucy/rethinkdb/build/packages/js/node_modules/bluebird/js/main/util.js:51:21)
    at Function.Promise$_successAdapter (/home/ssd2/mlucy/rethinkdb/build/packages/js/node_modules/bluebird/js/main/nodeify.js:47:11)
    at tryCatch1 (/home/ssd2/mlucy/rethinkdb/build/packages/js/node_modules/bluebird/js/main/util.js:43:21)
    at Promise$_callHandler [as _callHandler] (/home/ssd2/mlucy/rethinkdb/build/packages/js/node_modules/bluebird/js/main/promise.js:649:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/ssd2/mlucy/rethinkdb/build/packages/js/node_modules/bluebird/js/main/promise.js:663:18)
Oops, this shouldn't have happened:
runTest polyglot/changefeeds/basic.js/1 [None] [ReferenceError: None is not defined]
cleaning up table...

<<< end Failed: polyglot/changefeeds/basic.js
== Starting: polyglot/changefeeds/basic.rb1.9 (T+ 6.0 sec)
== Passed: polyglot/changefeeds/basic.py2.6 in 6.3 sec (T+ 9.1 sec)
>>> Failed polyglot/changefeeds/basic.rb1.9 with exit code 1 after 6.4 sec (T+ 12.8 sec)

SETUP ERROR: polyglot/changefeeds/basic.rb1.9/1
        BODY: tbl.get(0).changes().limit(1)
        EXPECTED: "[None]"
        FAILURE: uninitialized constant None

Failed 1 tests
Ruby: 18 of 19 tests passed. 1 tests failed.

<<< end Failed: polyglot/changefeeds/basic.rb1.9

== Failed 2 tests (of 3) in 14.98 seconds!
        polyglot/changefeeds/basic.js
        polyglot/changefeeds/basic.rb1.9
```

It appears to be rewriting `null` to `None` regardless of language.  Ideally it would rewrite it to the correct equivalent in all 3 languages, but if it can't do that I think it shouldn't rewrite it at all.

**Comment by mlucy** [comment]
Actually, oddly enough, the following works:
```
    - cd: tbl.get(0).changes().limit(1)
      ot: ([null])
```

**Comment by larkost** [comment]
This bit was written wrongly, partially because there seems to have been some mis-understanding about how YAML handles `null`/`nil`/`None`. It is [documented here](http://yaml.org/type/null.html), but the summary is that in YAML any of these is translated into `None` in Python from YAML:
```
  # empty
~
null
Null
NULL
```

We then have code that should take this `None` value, and replace it with the appropriate value. However, that code was looking for the string `null` in Ruby and Python, and in JavaScript was trying to translate it to `None`. We mostly go by falling through holes.

I have already fixed the code in `test-runner`, but now have to fix all the myriad variations in the YAML files.

**Comment by larkost** [comment]
This is up for review on CR 2130, from branch `larkost/3580-rewiting-null`

**Comment by larkost** [comment]
This has been merged into `next` with 9918c1a, and cherry picked into `v1.16.x` (to make testing easier) with 802331f.
